apps/web/lib/trpc/routers/api.ts

```diff
@@ -59,21 +59,46 @@ export const apiRouter = t.router({
             message: "unable to emit event to trigger.dev",
           });
         }
-        await db
-          .update(schema.apis)
-          .set({ state: "DELETION_IN_PROGRESS" })
-          .where(eq(schema.apis.id, input.apiId));
+        await db.transaction(async (tx) => {
+          await tx
+            .update(schema.apis)
+            .set({ state: "DELETION_IN_PROGRESS" })
+            .where(eq(schema.apis.id, input.apiId));
+          await tx.insert(schema.auditLogs).values({
+            id: newId("auditLog"),
+            time: new Date(),
+            workspaceId: api.workspaceId,
+            actorType: "user",
+            actorId: ctx.user.id,
+            event: "api.delete",
+            description: `API ${api.name} deleted`,
```

Comment:
My bad, this one doesn't delete, but only start the process.

```suggestion
            description: `API ${api.name} marked for deletion`,
```

Reply:
Changed